server/server.go

```diff
@@ -484,6 +496,12 @@ func (a *ArgoCDServer) Init(ctx context.Context) {
 // golang/protobuf).
 func (a *ArgoCDServer) Run(ctx context.Context, listeners *Listeners) {
 	a.userStateStorage.Init(ctx)
+
+	metricsServ := metrics.NewMetricsServer(a.MetricsHost, a.MetricsPort)
+	if a.RedisClient != nil {
+		cacheutil.CollectMetrics(a.RedisClient, metricsServ)
+	}
```

Comment:
Is there an important reason this got moved up to Init from Run? No concerns on my part, just want to know for posterity's sake. :-)

Reply:
That was a necessary move because I needed to provide the `MetricsServer` instance to the `HttpServer` in the `newHTTPServer()` function. I was also concerned about this move and while doing so, I double checked the code and I didn't find a negative side effect caused by it as related variables and usage seems to be well isolated.